### DISCUSSION TITLE
Installing from ISO - Failed @ the End - Hyper-V -VM

### DISCUSSION THREAD

**Discussion by NeoLife-iT** [Security-Onion-Solutions/securityonion#12525] [2.4] [answered]
### Version

2.4.50

### Installation Method

Security Onion ISO image

### Description

installation

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

12

### RAM

128

### Storage for /

128Gig

### Storage for /nsm

Don't know

### Network Traffic Collection

other (please provide detail below)

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

No, one or more services are failed (please provide detail below)

### Salt Status

No, there are no failures

### Logs

Yes, there are additional clues in /opt/so/log/ (please provide detail below)

### Detail

![3-7-2024 10-13-59 AM](https://github.com/Security-Onion-Solutions/securityonion/assets/36971330/b83c4488-e6b7-4b3c-a914-31a050287180)

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by NeoLife-iT** [comment]
Here is the installation options





![3-7-2024 11-13-40 AM](https://github.com/Security-Onion-Solutions/securityonion/assets/36971330/ae13db5a-5f84-4187-91e4-34b3cb5b5bad)


here is the error at the begging of the installation - it said can not resole host but i can ping the host


![3-7-2024 11-17-33 AM](https://github.com/Security-Onion-Solutions/securityonion/assets/36971330/0ecd9233-4a90-44ec-b7f3-bbf1bd014a94)

**Reply by NeoLife-iT** [reply]
it take awhile but here are the result from both of my test install.  one is the corp. environment and 1 in my home lab - FAILURED

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36971330/8e066500-5231-4045-a895-732f3198938c)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36971330/49db06b8-e1dc-42e0-9787-f2846192b700)

**Comment by NeoLife-iT** [comment]
here is the error at the end

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36971330/4a802c45-5bf4-4f94-ba11-d9460d6242a0)



Here is the errors logs as the screenshot said high lighted

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36971330/54f535ff-8487-46db-8fe1-36450af26853)


Any Suggestion for fix this error?

**Reply by cm-ops** [reply]
If you run the `so-preflight` script does it complete successfully?

**Reply by NeoLife-iT** [reply]
This is what it showed - Standalone - install option

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36971330/8370b2b1-e091-457b-90da-92ec4917ac76)

**Reply by NeoLife-iT** [reply]
The ISO checked out - looked ok

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36971330/edcd93e1-051c-479a-9c01-179e56423381)

**Reply by cm-ops** [reply]
Did `so-preflight` complete successfully?

**Reply by NeoLife-iT** [reply]
NO it said "FAILURE"


![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36971330/924bbef3-291e-4ecc-bdcc-c810d2ccd6eb)

**Reply by cm-ops** [reply]
What was the error(s) in `/root/preflight.log`?

**Comment by NeoLife-iT** [comment]
Does anyone successfully install SO with hyper-V & Using ISO?

**Comment by NeoLife-iT** [comment]
Here is the preflight.log

Looked like can not update apt

**Reply by NeoLife-iT** [reply]
[preflight.log](https://github.com/Security-Onion-Solutions/securityonion/files/14711827/preflight.log)

**Reply by NeoLife-iT** [reply]
The preflight.log show the apt-get command error but on Oracle Linux it is using the DNF command instead.  If this is badly cut ISO are you planning to release the update version soon?

**Reply by NeoLife-iT** [reply]
Latest version: 2.4.60 - ISO - Same Error - Failed Installed.  Do you test these ISO release from Outside your  Developer Environment?
I think it is the firewall setting issue.   what command to disable the firewall on the  ISO setup environments.  I am new to this distro so need some pointers to install from ISO - My Goal is to build a Hyper-V version - but if you can release that it would be great.  Save me a lot of trouble.

**Reply by TOoSmOotH** [reply]
We test on bare metal, Proxmox and VMware. There are users who also have this running on hyper v. Looking at your first attempt that you posted it was unable to resolve sigs.securityonion.net. The other you look to have mixed case in the name unless that is a 0.

You can also try a standard OL9 minimal install and then install over the network.

**Reply by NeoLife-iT** [reply]
I understood why it can not resolved FQDN.   This is related to sOs Linux Kernel not compiling or missing the VLAN module.  In A corporate network that using VLAN when running to this issue you can configured the Native VLAN for your installation..  My Hyper-V host is configured for VLAN so I have a fight on my hand to get this install.  

-I take your suggestion and will try the Network Installation route but I need to learn Oracle Linux which take me to this site. 
https://oracle-samples.github.io/oltrain/playlists/

- I am learning Linux from Scratch this week-end - It is a good refresher course.  This distro install interface looked familiar to me.  It is remind me of Suse Linux that I worked with many Years ago when I first start learning Linux.
- 
- As it is now for my installation note

Install sOs from ISO - Hyper-V Cluster Environment Do & Don't

1. No VLAN  - VM must be configure with Native VLAN or configure your installation to work with VLAN.  Simple check is configured network only option - ping www.google.com if it can resolved to IP and able to ping you are good to go

2. Work around this Issue: pending

**Reply by TOoSmOotH** [reply, marked as answer]
There is no SO specific kernel. We use the stock Oracle linux 9 kernel. My suggestion would be to get a minimal OL9 working before trying to make SO work in your environment.

**Reply by NeoLife-iT** [reply]
Update on Work Around

2. Update The hosts files with Entries Listed on SOS install Site

hosts
-------------------------------------------------------------------------------------------

127.0.0.1   s0s s0s.localdomain localhost localhost.localdomain localhost4 localhost4.
localdomain
::1   s0s s0s.localdomain localhost localhost.localdomain localhost6 localhost6.locald
omain6
# comments to here: my work around for sOs ISO install
#<IP_address> <fully_qualified_domain_name>
# more comments here: errors not able to resolve FQDN
185.199.110.133 raw.githubusercontent.com
13.33.21.113            sigs.securityonion.net
13.33.21.113            d15e1rs01gh6d2.cloudfront.net
192.30.255.164  ghcr.io
185.199.108.154 pkg-containers.githubusercontent.com
4.225.37.64             rules.emergingthreatspro.com
54.156.118.228  rules.emergingthreats.net
192.30.255.112  github.com
172.67.69.209           repo.securityonion.net
104.26.14.42            repo-alt.securityonion.net
149.137.137.254 so-repo-east.s3.us-east-005.backblazeb2.com
54.230.21.59            download.docker.com
18.164.154.51           repo.saltstack.com
104.154.207.153 geoip.elastic.co
142.250.72.155  storage.googleapis.com
104.18.139.9            www.snort.com
------------------------------------------------------------------------------------------------
[root@s0s ~]#
[root@s0s ~]# ping sigs.securityonion.net
PING sigs.securityonion.net (13.33.21.113) 56(84) bytes of data.
64 bytes from sigs.securityonion.net (13.33.21.113): icmp_seq=1 ttl=248 time=770 ms
64 bytes from sigs.securityonion.net (13.33.21.113): icmp_seq=2 ttl=248 time=311 ms
^C

3. Mount the iso - This is important - the iso auto eject on the first stage of installation

4. Run the setup from Command line - This work faster and I seem to have best result - sudo SecurityOnion/setup/so-setup iso

on my box :

[root@s0s ~]# cd SecurityOnion/setup/
[root@s0s setup]# ls
files            public_keys   so-functions.orig  so-setup      so-verify
install_scripts  so-functions  so-preflight       so-variables  so-whiptail
[root@s0s setup]# sh so-setup iso


-Installed Completed with error related to the cert

[ERROR   ] An exception occurred in this state: Traceback (most recent call last):
      Result: False
     Comment: One or more requisite failed: ca.pki_public_ca_crt
      Result: False
      Result: False
     Comment: One or more requisite failed: ca.pki_public_ca_crt
      Result: False
      Result: False
     Comment: One or more requisite failed: ssl.influxdb_crt
      Result: False
      Result: False
     Comment: One or more requisite failed: ssl.redis_crt
      Result: False
      Result: False
     Comment: One or more requisite failed: ssl.etc_elasticfleet_crt
      Result: False
      Result: False
     Comment: One or more requisite failed: ssl.etc_elasticfleet_logstash_crt
      Result: False
      Result: False
     Comment: One or more requisite failed: ssl.etc_elasticfleet_logstash_key
      Result: False
     Comment: One or more requisite failed: ssl.etc_elasticfleetlumberjack_crt
      Result: False
--More--(15%)

👍 

I was able to get to  the Site with the login screen...somehow my login password didnt work so I will reinstall :-)

**Reply by NeoLife-iT** [reply]
Re-Install for the 40 times :-).  That what I've doing for the last 2 Weeks

Looking Good so Far.  

 I also have a theory why it not install normally ....FAILURED.....Stay Tune

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36971330/28a15d89-9721-43ef-98f1-98c3ebdf8407)

**Reply by NeoLife-iT** [reply]
After trying to All Day today on a Dell Vertex Blade Server ..using a physical blade trying to installed the Plain OL9...all failed...I took 1 step back and move the VM to Others Hyper-V Production Server...this Cluster is Standard node and SAN...not like the Vertex.......S0B!!!!!!! it worked!!!!!!  

--Now I will need your help deploy SOS on Google Cloud...I have installed the instance 10 days ago but it have only 1 nic :-)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36971330/a541a3c4-d1e0-441c-9c42-8d3794a93a82)

**Reply by NeoLife-iT** [reply]
This installed with error.  - error log is attached.

- so-elastic-fleet is not started & attempt to start resulting in complying errors

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36971330/3302bf38-1115-42fc-9919-2a401f285654)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36971330/be7e0944-aebf-4435-8bc8-1646069cec2b)



-

**Reply by NeoLife-iT** [reply]
So after running for a week all was fine until I reboot...it said in grid page that my installed need to reboot.  Reboot and waited for 1 hours ...can not access the web interface. login and check everything is off - attempted to start individual services failed so have to scrap everything and resinstall from scatch.

This time I installed it as distributed network manager --ran and surrived a reboot
-After added other node to this installation such as forwarder, search, receiver mode.  all this node does not able to ping the host or outisde server..  But they were able to joined the nothing work after that...so I scrapped the entire installation

Next installation number 70 LOL

This time as Manager and Search mode...it run and survived reboot so far but I am not adding other node to it.

Can I operate this searchmanager node alone?

**Comment by NeoLife-iT** [comment]
[errors.log](https://github.com/Security-Onion-Solutions/securityonion/files/14775869/errors.log)

**Reply by NeoLife-iT** [reply]
Rerun the setup and used my own on premise DNS Server - See Below

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36971330/9822eb86-560a-4df3-9f4d-dfc272f9a259)

I can login to the manager site

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36971330/76fc8faf-0626-4321-9661-4f3f827b6f0e)

 and fleet site

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36971330/6b4ed3b7-9da6-4767-ad97-b7d2f75e4061)

**Reply by NeoLife-iT** [reply]
Now I have the On-Premise installed and operational.  We're moving on to the Cloud Version
Question:  I have the App Instance Running on Google Cloud purchased from Market Play.  
1. Can not connect to it from the External IP - not sure SOS is run
2. Can that Instance connect to the Data from the On-Premise Installed?  
3. How is the Google Cloud Version Different to the On-Premise Version?  
4. Where do I go for Help with the Google Cloud Version?

Thanks for All your help

**Reply by NeoLife-iT** [reply]
Google Cloud Instance
it look like everything is running.  how do i login to the web interface?

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36971330/b9b7ba25-f5f5-4607-8d16-9ddacc5e29c5)

**Reply by NeoLife-iT** [reply]
This might be useful to other having similar install issue

**Reply by NeoLife-iT** [reply]
Question:  My First installed was Standalone Mode Install  Since Then I had added to the StandAlone Manager

1. Forward Sensor Node
2. Searchn Node
3. Receiver Node

1. Do I need to add Manger Node? or StandAlone Manager is already the Manager Node when I add other node to this deployment?

2. Can I add the Manager Node and then Decomission the StandAlone Installation?

Picture below show my Grid Members....Do they actually work together now?  how do I know?

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36971330/bde0ad06-24a0-4624-8466-465bbeade144)
